department-of-veterans-affairs/abd-vro
# Feature - add controller for domain-cc

Part 1 of 3 for [Figure out how to talk from CC to VRO gateway #41](https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/41)

This code defines a new controller which accepts a string to indicate different endpoints for CC API. Currently, logging the endpoint. Next steps will include communicating w/ RabbitMQ.

## What was the problem?
A thin java controller code is needed to make API requests available to a new Contention Classification service.

Associated tickets or Slack threads:

- [#41 on vagov-claim-classification](https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/41)

## How does this fix it?[^1]
A developer running abd-vro can invoke POST api calls to `https://localhost:8080/domain-cc/foo`, `https://localhost:8080/domain-cc/bar`, etc

## How to test this PR
- build and run vro-app-1
- submit post request either from swagger docs on http://localhost:8080/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config#/ or a tool like Postman
- observer output to vro-app-1's stdout


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
